src/query_router.rs

```diff
@@ -114,7 +115,52 @@ impl QueryRouter {
 
         let code = message_cursor.get_u8() as char;
 
-        // Only simple protocol supported for commands.
+        // Check for any sharding regex matches in any queries
+        match code as char {
+            // For Parse and Query messages peek to see if they specify a shard_id as a comment early in the statement
```

Comment:
I think we can avoid this entire branch when the number of shards is 1, we can get that from `self.pool_settings.shards`. This will make this check even cheaper for the base, unsharded use case

Reply:
Scratch that, the lack of the regex in the confs should be enough of a gate